core-library/src/main/java/org/silverpeas/core/security/authorization/PublicationAccessController.java

```diff
@@ -288,7 +296,14 @@ private boolean fillTopicTrackerNodeRoles(final Set<SilverpeasRole> userRoles,
   private boolean fillTopicTrackerAliasRoles(final Set<SilverpeasRole> userRoles,
       final AccessControlContext context, final String userId, final PublicationPK pubPk) {
     final Set<AccessControlOperation> operations = context.getOperations();
-    if (!isPersistActionFrom(operations)) {
+    // Rights on alias are verified if no persist action detected and if it exists a main
+    // location on another component instance than the one hosting pubPk which is not removed
+    if (!isPersistActionFrom(operations) &&
+        getDataManager(context).getPublicationMainLocationSupplier(pubPk)
+            .get()
+            .filter(l -> !l.getInstanceId().equals(pubPk.getInstanceId()))
+            .map(l -> !isComponentInstanceRemoved(l.getInstanceId(), context))
+            .orElse(true)) {
```

Comment:
Put the result of this check in an understandable boolean variable and then uses it in the if-condition along with the isPersistActionFrom method call

Reply:
I have not wait for your answer, I have finally implemented the private method.